src/domain/common/entities/DataElement.ts

```diff
@@ -65,7 +65,12 @@ type CategoryCombos = {
     categoryOptionCombos: {
         id: Id;
         name: string;
+        formName: string | undefined;
         shortName: string | undefined;
+        categoryOptions?: {
+            id: Id;
+            code: string;
+        }[];
```

Comment:
Not sure why we need these `categoryOptions?`. It's used in some view model, but where is being filled? If I remove the property and the usage, TS does not complain, so this should mean nobody is filling it.

Reply:
Yes, but where is the property being set?

Reply:
it's filled in `Dhis2DataElement.ts` line 134

Reply:
Can you paste the link? at https://github.com/EyeSeeTea/d2-autogen-forms/blob/feat/change-grid-with-cat-option-combos-colTitle-86944kn4z/src/data/common/Dhis2DataElement.ts#L134 I don't see the set.

Reply:
> dumebi.  okay so it’s set on line 136: const categoryOptionCombos = dataElement.categoryCombo.categoryOptionCombos; the value of categoryOptions is gotten directly from the d2-response`

That's the point, thanks.

The general rule here is not to use optional properties unless strictly necessary.